### ISSUE TITLE
Password generator: Triple-click does not select password

### ISSUE THREAD

**Issue by kleinfreund** [bitwarden/browser#896]
When generating a new password, triple-clicking and double-clicking do not work as expected. Atleast on Windows and Linux, the behavior for double-clicking and triple-clicking text is roughly as described here:

**Double click**: Depending on the location of the caret, the string under it will be selected up to the next left and right boundaries (e.g. as indicated by a space, dash, ... characters). When the caret is hovered over a word, double-clicking results in the full word being selected.

**Triple click**: After performing a double-click and selecting a word (for example), a quick third click expands the selection further to left and right boundaries indicated by a new line or the end of the string.

Although the exact algorithm for determining what is selected varies from OS to OS, I believe the general behavior is similar on all major operating systems.

TL;DR: **In particular in the field where the generated password is displayed, triple-clicking should select the full generated password.**

---

At the very least, use `user-select: all` to make a single click into the field select the whole password:

```css
app-password-generator .password-block {
  user-select: all; /* vendor prefixes required */
}
```

I wonder why the single characters are wrapped in span elements because that’s causing that issue.

---

The following file is responsible for the individual `span` elements:

[`jslib/src/angular/pipes/color-password.pipe.ts`](https://github.com/bitwarden/jslib/blob/master/src/angular/pipes/color-password.pipe.ts)

**Comment by kleinfreund** [comment]
I’d like to propose a change and submit a pull request if you’re interested:

I want to change the algorithm in [jslib/src/angular/pipes/color-password.pipe.ts](https://github.com/bitwarden/jslib/blob/master/src/angular/pipes/color-password.pipe.ts) to only group special characters together. Words should not be split into multiple spans.

**Comment by kspearrin** [comment]
Sure. I'd be open to seeing what you propose in a PR.

**Comment by kleinfreund** [comment]
Just a note: My initial assessment of the cause of this issue was wrong. The problem wasn’t the individual spans but the usage of CSS flexbox.

**Comment by kspearrin** [comment]
While this would work for the password generator screen, it breaks long password displays on the login view page.
